src/main/kotlin/pl/droidsonroids/gradle/jacoco/testkit/JacocoTestKitExtension.kt

```diff
@@ -5,28 +5,36 @@ import org.gradle.api.Task
 import org.gradle.api.provider.Provider
 import org.gradle.api.tasks.TaskProvider
 import org.gradle.testing.jacoco.plugins.JacocoTaskExtension
+import org.gradle.testing.jacoco.tasks.JacocoReport
 import java.io.File
 
 open class JacocoTestKitExtension(private val project: Project) {
     val jacocoRuntimePathProvider: Provider<String> = project.provider { project.configurations.getByName(Configurations.jacocoRuntime).asPath }
 
     fun applyTo(configurationRuntime: String, taskProvider: TaskProvider<Task>) {
         with(project) {
+            val destinationFile = taskProvider.map { task ->
+                val extension = task.extensions.getByType(JacocoTaskExtension::class.java)
+                val file = checkNotNull(extension.destinationFile) { "destinationFile is missing on task $name jacoco extension" }
+
+                File(file.parentFile, "${file.nameWithoutExtension}-gradle-runner.${file.extension}")
+            }.forUseAtConfigurationTime()
+
             val jacocoTestKitPropertiesTask = tasks.register(
                 generatePropertiesTaskName(taskProvider.name),
                 GenerateJaCoCoTestKitProperties::class.java
             ) {
-                it.outputFile = File(buildDir, "testkit/${taskProvider.name}/testkit-gradle.properties")
-                it.destinationFile.set(
-                    taskProvider.map {
-                        task -> task.extensions.getByType(JacocoTaskExtension::class.java).destinationFile!!.path
-                    }.get()
-                )
+                it.outputFile = File(buildDir, "/testkit/${taskProvider.name}/testkit-gradle.properties")
+                it.destinationFile.set(destinationFile.get().path) // .get() breaks the circular dependency
```

Comment:
I'm not sure if this `.get()` was intentional but my natural approach to this will be:
```kotlin
it.destinationFile.set(destinationFile.map(File::getPath))
``` 

The problem is that also adds a dependency from `jacocoTestKitPropertiesTask` to `taskProvider` and as later we do:
```kotlin
taskProvider.configure { it.dependsOn(jacocoTestKitPropertiesTask) }
```
a circular dependency is form.

Reply:
There is even a comment in this line `// .get() breaks the circular dependency`

Reply:
The comment was to state that without a `.get()` that breaks the dependency inference of the Property API, a dependency loop is made.

Anyway I've just removed it.